File: backend/app/cypher/schemas.py
LIST = """
MATCH (s:NodeTypeSchema {project_id: $pid})
RETURN s ORDER BY s.type_key
"""

GET = """
MATCH (s:NodeTypeSchema {project_id: $pid, type_key: $type_key})
RETURN s
"""

CREATE = """
CREATE (s:NodeTypeSchema {
  id: $id, project_id: $pid, type_key: $type_key, display_name: $display_name,
  fields: $fields, created_at: datetime(), updated_at: datetime()
})
RETURN s
"""

UPDATE = """
MATCH (s:NodeTypeSchema {project_id: $pid, type_key: $type_key})
SET s.display_name = $display_name, s.fields = $fields, s.updated_at = datetime()
RETURN s
"""

DELETE = """
MATCH (s:NodeTypeSchema {project_id: $pid, type_key: $type_key})
DELETE s
"""

COUNT_NODES_OF_TYPE = """
MATCH (n:LineageNode {project_id: $pid, type: $type_key})
RETURN count(n) AS cnt
"""

LIST_NODES_OF_TYPE = """
MATCH (n:LineageNode {project_id: $pid, type: $type_key})
RETURN n
"""

File: frontend/src/api/schemas.ts
import { http } from "./client"
import type { NodeTypeSchema, SchemaField } from "@/types/graph"

export const schemasApi = {
  list: (pid: number) => http.get(`/projects/${pid}/schemas`) as unknown as Promise<NodeTypeSchema[]>,
  create: (pid: number, body: { type_key: string; display_name: string; fields: SchemaField[] }) =>
    http.post(`/projects/${pid}/schemas`, body) as unknown as Promise<NodeTypeSchema>,
  update: (pid: number, typeKey: string, body: { display_name?: string; fields?: SchemaField[] }) =>
    http.put(`/projects/${pid}/schemas/${typeKey}`, body) as unknown as Promise<NodeTypeSchema>,
  remove: (pid: number, typeKey: string) =>
    http.delete(`/projects/${pid}/schemas/${typeKey}`) as unknown as Promise<void>,
}
